compaction/compaction_manager.cc

```diff
@@ -1428,7 +1425,7 @@ future<> compaction_manager::perform_sstable_scrub(replica::table* t, sstables::
 }
 
 void compaction_manager::add(replica::table* t) {
```

Comment:
we'll need to add / remove individual compaction_group:s
as we'd want to split / merge them dynamically.

The bottom line is that the compaction_manager api should switch almost entirely to using table_state / compaction_group (plus the ability to stop compaction by table since it's a common use case that currently requires O(n) traversal of all all tasks)

Reply:
In the end compation/ shouldn't include anything from replica/.

Reply:
That's what we'll achieve by the end of this series.